ai-service/src/ai_service/chunking/strategies.py

```
@@ -0,0 +1,64 @@
+        # Skip empty/whitespace-only files
+        nonempty = sum(1 for line in lines if line.strip())
+        if nonempty == 0:
+            return []
+        return [_add_file_context(file_path, content, "complete-file")]
+
+    chunks: list[str] = []
+    chunk_size = 30  # Target lines per chunk
+    overlap = 5  # Lines to overlap between chunks for context continuity
+    step = chunk_size - overlap
+    if step <= 0:
+        raise ValueError("chunk_size must be greater than overlap")
```

Ruff (0.12.2)

Avoid specifying long messages outside the exception class

(TRY003)